processing/src/main/java/org/apache/druid/query/operator/WindowOperatorQueryQueryToolChest.java

```diff
@@ -116,6 +128,36 @@ public Sequence<Object[]> resultsAsArrays(
     return (Sequence) resultSequence;
   }
 
+  @Override
+  public Optional<Sequence<FrameSignaturePair>> resultsAsFrames(
+      WindowOperatorQuery query,
+      Sequence<RowsAndColumns> resultSequence,
+      MemoryAllocatorFactory memoryAllocatorFactory,
+      boolean useNestedForUnknownTypes
+  )
+  {
+    RowSignature rowSignature = resultArraySignature(query);
+    RowSignature modifiedRowSignature = useNestedForUnknownTypes
+                                        ? FrameWriterUtils.replaceUnknownTypesWithNestedColumns(rowSignature)
+                                        : rowSignature;
```

Comment:
this method has `result` in its name; but it seems like the `result*Signature` needed some fixups/checks; isn't this `modifiedRowSignature` the right `signature` for the `query` ?

Reply:
We had a problem of unknown types in the subqueries which is largely resolved now. Theoretically, each primitive/array type can be serialized as JSON type. Therefore if somebody sets `useNestedForUnknownColumnTypes`, we replace columns without a signature with JSON type, so that they can be serialized as frames. 
It is an undocumented parameter, and is only applicable for `resultsAsFrames` since that relies on types being present.

Reply:
Also, it was mostly experimental and for dev purpose only because it mangled the data types.